core/node/fee_model/src/l1_gas_price/main_node_fetcher.rs

```diff
@@ -73,6 +99,29 @@ impl MainNodeFeeParamsFetcher {
 
 impl BatchFeeModelInputProvider for MainNodeFeeParamsFetcher {
     fn get_fee_model_params(&self) -> FeeParams {
-        *self.main_node_fee_params.read().unwrap()
+        let fee_params = *self.main_node_fee_params.read().unwrap();
+        let batch_fee_input = *self.main_node_batch_fee_input.read().unwrap();
+        if batch_fee_input.is_none() {
+            return fee_params;
+        }
+        let batch_fee_input = batch_fee_input.unwrap();
```

Comment:
nit: you can use `let Some() ... else ...` and avoid `unwrap`

Reply:
I agree, but at the cost of extra indentation, so I'd prefer to keep it that way. It's one of very few rust-specific syntactic sugars that Imo do more harm than good in terms of readability.

Reply:
What extra indentation? I think what @perekopskiy is suggesting is doing this:
```suggestion
        let Some(batch_fee_input) = batch_fee_input else {
            return fee_params;
        }
```
Which is IMO more idiomatic

Reply:
nvm, my brain malfunctioned, thanks!